### PR TITLE
Clean Up some Code

### PR DESCRIPTION
**Pull request by jokoho48**
Some small Code Clean Up

Replace QUOTE(FUNC()) with QFUNC()

fnc_sendDisplayMessageTo Change von BIS_fnc_Param to New Params command

### CODE REVIEW

**Comment by jonpas** [comment]
Clean up == :+1:

**Comment by jokoho48** [comment]
found 24 hits for BIS_fnc_Param Currently Changing

File: addons/common/functions/fnc_unloadPersonLocal.sqf [hunk 1]
```diff
@@ -13,7 +13,7 @@
 //#define DEBUG_MODE_FULL
 #include "script_component.hpp"
 
-#define GROUP_SWITCH_ID QUOTE(FUNC(loadPerson))
+#define GROUP_SWITCH_ID QFUNC(loadPerson)
```
Comment (SilentSpike) [inline addons/common/functions/fnc_unloadPersonLocal.sqf]: These are not technically the same thing, but it probably doesn't matter.

Comment (jokoho48) [inline addons/common/functions/fnc_unloadPersonLocal.sqf]: Return is both "ace_ADDON_fnc_function" what is ther different?

Comment (SilentSpike) [inline addons/common/functions/fnc_unloadPersonLocal.sqf]: Check out this file here: https://github.com/acemod/ACE3/blob/master/addons/main/script_debug.hpp

Comment (jokoho48) [inline addons/common/functions/fnc_unloadPersonLocal.sqf]: hmm ok thats for Debug hmm

**Comment by jokoho48** [comment]
More Improvements are incomming after i take a nap

**Comment by PabstMirror** [comment]
This is pretty big change to everything by swapping all the PARAMS_X() with the new `params`.

Ref #1821
@SilentSpike has pushed changes https://github.com/CBATeam/CBA_A3/pull/93 to CBA that would do some of this.

But, one issue this does solve is all the variadic functions E.G.
```
params ["_unit", "_animation", ["_priority", 0], ["_force", false]];
```
Which have been done by hand very nicely here.

I think macroing that would be very ugly.

Biggest downside to this is we lose the auto `TRACE_X` from the `PARAM_X` which has made debugging easier.  But we can always add these back in.

**Comment by MikeMatrix** [comment]
@PabstMirror I agree with this in general, especially the point about `TRACE_X` being missing.
Problem I would still see is that currently from what I've seen, parameter handling is so inconsistent, that even if `TRACE_X` would be there, a quite significant amount of functions still wouldn't display any information, or partial information, because some of it was handled through some custom code anyways.
I think introducing some consistent way is a necessity, but I'm unsure if this can be solved using macro's, considering the varying nature of parameters a lot of times.

Maybe we introduce calling `TRACE_X` right after parameterization is completed.

**Comment by SilentSpike** [comment]
Keep in mind the params commands do have a debugging tool built into them:

>In addition to simple parsing directly into variables, input can be tested in case it is undefined, of the wrong type or of the wrong size (if array) and substituted if necessary with default values. Substitutes can also be logged into .rpt file, if allowFunctionsLog param in description.ext is set to 1.

**Comment by MikeMatrix** [comment]
So do you think it wouldn't even be necessary to have a `TRACE_X` in the parameterization?

**Comment by SilentSpike** [comment]
That said, it was previously agreed that we wouldn't bother to replace every instance of the params macro with the new commands. At least not in one big change like this: #1821

**Comment by MikeMatrix** [comment]
@SilentSpike The initial idea wasn't really to replace those, but to remove those cases where the method used was a manual extraction completely or partially. Since this meant we were going through files anyways, it seemed logical to adjust things to a consistent fashion.
We could simply touch the cases that are explicitly using some form of custom written code for parameterization and leave the rest alone if that's more preferred.

**Comment by Glowbal** [comment]
This is to big to review effectively on github. It might be best if you split off any future work related to this on a separate branch and create a PR for that, so it's in smaller pieces.

**Comment by jokoho48** [comment]
i could split this in separate PR for every PBO one if this is ok?

**Comment by MikeMatrix** [comment]
Incoming PRs

**Pull request closed without merging**